File: CSharp/BinarySearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BinarySearch
{
    class Program
    {
        static void Main(string[] args)
        {
            var arr = new int[10] { 4, 6, 15, 45, 55, 69, 89, 120, 145, 166 };

            Console.WriteLine(BinarySearch(arr, 55)); // true
            Console.WriteLine(BinarySearch(arr, 200)); // false

            Console.ReadKey();
        }

        public static bool BinarySearch(int[] arr, int item)
        {
            var low = 0;
            var high = arr.Length - 1;

            while (low <= high)
            {
                var mid = (low + high)/2;
                if (arr[mid] == item)
                    return true;
                if (item < arr[mid])
                    high = mid - 1;
                else
                    low = mid + 1;
            }

            return false;
        }
    }
}

File: PYTHON/BinarySearch.py
def binary_search(list, item):
    low = 0
    high = len(list)-1
    
    while  low <= high:
        mid = (low + high)
        guess = list[mid]
        if guess == item:
            return mid
        if guess > item:
            high = mid -1
        else:
            low = mid - 1
    return None
    
my_list = [1, 3, 4, 5, 7, 9, 12]

print (binary_search(my_list, 3))
print (binary_search(my_list, -1)) 
